chain/chain/src/chain.rs

```diff
@@ -4758,7 +4757,7 @@ impl Chain {
         receipts: Vec<Receipt>,
         shard_layout: &ShardLayout,
     ) -> HashMap<ShardId, Vec<Receipt>> {
-        let mut result = HashMap::with_capacity(shard_layout.num_shards() as usize);
+        let mut result = HashMap::new();
```

Comment:
assuming that the number of receipts is significantly larger than number of shards, I think pre-populating the number of slots in the hashmap is a premature optimisation.